src/index.ts

```
@@ -468,13 +524,21 @@ function handlePaywallEvent(event: HeliumPaywallEvent) {
     case 'paywallOpenFailed':
       paywallEventHandlers = undefined;
       const unavailableReason = event.paywallUnavailableReason;
+      // Clear before invoking: the callback may call presentUpsell again, and clearing
+      // afterwards would drop the new presentation's handlers.
+      const onPaywallUnavailable = presentOnPaywallUnavailable;
```

Biome (2.5.8)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

(lint/correctness/noSwitchDeclarations)